### PR TITLE
iterators: filter predicate returns bool

### PR DESCRIPTION
**Pull request by ZanderBrown**
Unless I'm greatly mistaken the predicate for filter is the same as find, rather than map's transformation func

### CODE REVIEW

File: discord-stubs/iterators.pyi [hunk 1]
```diff
@@ -22,7 +22,7 @@ class _AsyncIterator(Generic[_IT]):
         self, func: Callable[[_IT], Union[_IT, Coroutine[Any, Any, _IT]]]
     ) -> _MappedAsyncIterator[_IT]: ...
     def filter(
-        self, predicate: Callable[[_IT], Union[_IT, Coroutine[Any, Any, _IT]]]
+        self, predicate: Callable[[_IT], bool]
```
Comment (bryanforbes) [inline discord-stubs/iterators.pyi:25]: The predicate can return a `bool` or a coroutine resolving in a `bool`, so this should be `Union[bool, Coroutine[Any, Any, bool]]`.

**Comment by bryanforbes** [comment]
@ZanderBrown you're absolutely right. Good catch! I'll merge this as soon as my comments are addressed. Thanks!

File: discord-stubs/iterators.pyi (hunk 1 above)
Comment (ZanderBrown) [reply discord-stubs/iterators.pyi:25]: Ah indeed, that implies find is wrong as well

Comment (bryanforbes) [reply discord-stubs/iterators.pyi]: Yep. I was going to fix it after I merged your PR, but you took care of it. Thanks!! :)

**Comment by ZanderBrown** [comment]
Done

Must admit it confused me a bit when mypy reported that one :-)

**Pull request merged**